src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/TimeDerivative.hpp

```diff
@@ -318,7 +318,15 @@ struct ComputeTimeDerivImpl<
           if constexpr (not tmpl::list_contains_v<
                             tmpl::list<GrmhdSourceTags...>, GrmhdDtTags>) {
             // Zero GRMHD tags that don't have sources.
-            dt_var[i] = 0.0;
+            //
+            // Since dt(u) is independent of `dim`, and `dim` is really part
+            // of the `d_dim F^dim` computation, we only need to set a zero
```

Comment:
where does this `d_dim F^dim` computation happen? I didn't find `F^dim` in this file, and don't see `dim` passed to a function in this loop, so I'm not sure where to look

Reply:
dim is passed to `evolution::dg::subcell::add_cartesian_flux_divergence` on line 332 which does that computation, *adding* to `dt_var`. 

However, any variable that doesn't have a source term (the if constexpr on line 318) has been passed to this function in an uninitialized state.  Therefore it should be set to zero before being passed into the first call of `add_cartesian_flux_divergence`.  The bug had the effect that only the highest dim contributed to the flux divergence instead of all the dimensions...

Reply:
`evolution::dg::subcell::add_cartesian_flux_divergence(`

`dim` is used on lines 306, 308, and 310, which are then captured by the lambda.

Reply:
Ah I think what I was missing is "Zero GRMHD tags that don't have sources **in the first iteration of the loop over `dim`**" or something like that. Maybe add that to the comment?

Reply:
Isn't that exactly what `we only need to set a zero source term when dim is zero.` means?

Reply:
I needed while to understand that this refers to the first iteration of the loop, with subsequent iterations adding to the value 🤷‍♂️ I'm not blocking this, just tried to understand what was happening here

Reply:
I've rebased and reworded the sentences. Hopefully it's clearer now :)